src/main/java/org/icgc/argo/program_service/config/AppConfig.java

```diff
@@ -0,0 +1,91 @@
+package org.icgc.argo.program_service.config;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.velocity.app.VelocityEngine;
+import org.icgc.argo.program_service.utils.NoOpJavaMailSender;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.annotation.Validated;
+
+import javax.validation.constraints.NotNull;
+import java.util.Properties;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * Ego external configuration, served as metadata for application.yml
+ */
+@Slf4j
+@Component
+@ConfigurationProperties(prefix=AppConfig.APP)
+@Validated
+@Setter @Getter
+public class AppConfig {
```

Comment:
For convention, it should be called properties, right?
[Doc](https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-external-config.html#boot-features-external-config-typesafe-configuration-properties)
[Source Code snippet 1](https://github.com/spring-projects/spring-boot/blob/17aaf26590aadfc24b58be72a985ddbbe1611742/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jersey/JerseyProperties.java)
[Source Code snippet 2](https://github.com/spring-projects/spring-boot/blob/36c1c051b8106371d37f8a747cea0e369a8e5c84/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/hateoas/HateoasProperties.java)
[Source Code snippet 3](https://github.com/spring-projects/spring-boot/blob/36c1c051b8106371d37f8a747cea0e369a8e5c84/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/reactive/WebFluxProperties.java)

Reply:
make sense. I will also separate out properties from bean definitions. Beans will be defined in a *Config class so there is a common place where beans are constructed, and properties classes will only contain properties